src/Extensions/Hosting.Services.Web/Middlewares/ActivityEnrichmentMiddleware.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
+{
+	/// <summary>
+	/// Enrich request activity with Result, SubType and Methadata
+	/// </summary>
+	public class ActivityEnrichmentMiddleware
+	{
+		/// <summary>
+		/// Invoke middleware
+		/// </summary>
+		public async Task InvokeAsync(HttpContext context)
+		{
+			await m_next(context).ConfigureAwait(false);
+
+			Activity? activity = Activity.Current;
+
+			if (activity == null)
+			{
+				return;
+			}
+
+			int statusCode = context.Response.StatusCode;
+
+			activity.SetResult(GetResult(statusCode))
+				.SetSubType($"{context.Request.Scheme}:{context.Connection.LocalPort}")
+				.SetMetadata(statusCode.ToString());
+		}
+
+		internal ActivityEnrichmentMiddleware(RequestDelegate next) => m_next = next;
```

Comment:
>ActivityEnrichmentMiddleware [](start = 11, length = 28)

From readability perspective, to keep it aligned with the rest of the code, would it be better to have the constructor as first?

Reply:
Sure I think it would be better like this also, I was worried that less accessible member will be upper then more accessible, but I'm fine with changing it.